### ISSUE TITLE
Wrong triangulation for a specific dataset in devel 0.4.0

### ISSUE THREAD

**Issue by Jean-Romain** [davidcsterratt/geometry#11]
Hi again,

I just spent so much time to figure out why a bug appeared in my package but actually the bug occurs since I installed the devel version of geometry to check bug #10. 

I'm not able to reproduce the bug with dummy data such as randomly spread points so I shared you my data in Rdata file: [points.Rdata.zip](https://github.com/davidcsterratt/geometry/files/979314/points.Rdata.zip)

```r
library(geometry)

load("~/Bureau/points.Rdata")

t = delaunayn(cbind(x,y))

plot(x,y, cex = 0.1, col = "red")
trimesh(t, x, y, add = T)
```
As you can see below the triangulation is completely wrong. Removing v0.4.0 and reinstalling 0.3.6 everything is fine. I tryied to reproduce the bug with random points but each test leaded to a correct triangulation.

![trifail](https://cloud.githubusercontent.com/assets/3872279/25749372/42d058b2-317c-11e7-87ee-e904611202a6.png)

**Comment by davidcsterratt** [comment]
Nice (or nasty) catch! The immediate problem seems to be the `Qz` option, which was added in this commit: https://github.com/davidcsterratt/geometry/commit/8a1d495a0b70d355ed83d2e7cf2ba816b46a2854

Try
`t = delaunayn(cbind(x,y), options="Qt Qc")`
and you should get the right result.

I added the `Qz` option for compatibility with octave and scipy and because, IIRC, it fixes the edge case in issue #10. I'll think more about this...

**Comment by Jean-Romain** [comment]
Well I don't really have an opinion on this question. Maybe the original code was better (without Qz) since co spherical points is a very specific case. I mean it is ok to add an option Qz for a specific case but less intuitive to add two options Qt Qc because of some kind of random fails.

**Comment by davidcsterratt** [comment]
I've gone upstream with this - I'd rather understand why this behaviour occurs before making a decision about whether to drop the Qz option.

**Comment by cbbarber** [comment]
This will be fixed in Qhull 2019.1.   See my note for #25 'Qz option can lead...'

Thanks Jean-Romain for reporting this problem, and David for not letting it go.

**Comment by davidcsterratt** [comment]
This particular issue has been fixed by the upgrade to Qhull 8.0.2 (Issue #37). The patch above adds a test based on this example to avoid regressions.
